frontend/public/components/app.jsx

```diff
@@ -192,80 +193,94 @@ class App_ extends React.PureComponent {
+  getCachedResources()
+    .then((resources) => {
+      if (resources) {
+        store.dispatch(receivedResources(resources));
+      }
+      // Still perform discovery to refresh the cache.
+      startDiscovery();
+    })
+    .catch(startDiscovery);
 
-// Global timer to ensure all <Timestamp> components update in sync
-setInterval(() => store.dispatch(UIActions.updateTimestamps(Date.now())), 10000);
+  store.dispatch(detectFeatures());
 
-// Fetch swagger on load if it's stale.
-fetchSwagger();
+  // Global timer to ensure all <Timestamp> components update in sync
+  setInterval(() => store.dispatch(UIActions.updateTimestamps(Date.now())), 10000);
 
-// Used by GUI tests to check for unhandled exceptions
-window.windowError = null;
-window.onerror = (message, source, lineno, colno, error) => {
-  // eslint-disable-next-line no-console
-  console.error('Uncaught error', error);
-  window.windowError = error;
-};
-window.onunhandledrejection = (promiseRejectionEvent) => {
-  // eslint-disable-next-line no-console
-  console.error('Unhandled promise rejection', promiseRejectionEvent);
-  window.windowError = promiseRejectionEvent;
-};
+  // Fetch swagger on load if it's stale.
+  fetchSwagger();
 
-if ('serviceWorker' in navigator) {
-  if (window.SERVER_FLAGS.loadTestFactor > 1) {
-    // eslint-disable-next-line import/no-unresolved
-    import('file-loader?name=load-test.sw.js!../load-test.sw.js')
-      .then(() => navigator.serviceWorker.register('/load-test.sw.js'))
-      .then(
-        () =>
-          new Promise((r) =>
-            navigator.serviceWorker.controller
-              ? r()
-              : navigator.serviceWorker.addEventListener('controllerchange', () => r()),
-          ),
-      )
-      .then(() =>
-        navigator.serviceWorker.controller.postMessage({
-          topic: 'setFactor',
-          value: window.SERVER_FLAGS.loadTestFactor,
-        }),
-      );
-  } else {
-    navigator.serviceWorker
-      .getRegistrations()
-      .then((registrations) => registrations.forEach((reg) => reg.unregister()))
-      // eslint-disable-next-line no-console
-      .catch((e) => console.warn('Error unregistering service workers', e));
+  // Used by GUI tests to check for unhandled exceptions
+  window.windowError = null;
+  window.onerror = (message, source, lineno, colno, error) => {
+    // eslint-disable-next-line no-console
+    console.error('Uncaught error', error);
+    window.windowError = error;
+  };
+  window.onunhandledrejection = (promiseRejectionEvent) => {
+    // eslint-disable-next-line no-console
+    console.error('Unhandled promise rejection', promiseRejectionEvent);
+    window.windowError = promiseRejectionEvent;
+  };
+
+  // Used by GUI tests to check for unhandled exceptions
+  window.windowError = false;
+  window.onerror = window.onunhandledrejection = (e) => {
+    // eslint-disable-next-line no-console
+    console.error('Uncaught error', e);
+    window.windowError = e || true;
+  };
```

Comment:
Looks like a mistake rebasing here. You can remove these lines.

Reply:
Sorry, should be fixed now.